core/api.go

```diff
@@ -77,6 +77,7 @@ type API struct {
 	remoteGenerator *rpc.Client
 	indexTxs        bool
 	forwardUsingTLS bool
+	internalClient  *http.Client
```

Comment:
Two small bikeshed suggestions:

- call this httpClient
- add an end-of-line comment `// normally nil; only set in tests`